acceptance/tests/partitions/partitions_test.go

```diff
@@ -86,151 +86,134 @@ func TestPartitions(t *testing.T) {
-				logger.Logf(t, "retrieving partition service to determine external IP for servers")
-				partitionsSvc, err := serverClusterContext.KubernetesClient(t).CoreV1().Services(serverClusterContext.KubectlOptions(t).Namespace).Get(ctx, partitionServiceName, metav1.GetOptions{})
-				require.NoError(t, err)
-				partitionSvcIP = partitionsSvc.Status.LoadBalancer.Ingress[0].Hostname
-			} else {
+			var partitionSvcAddress string
+			if cfg.UseKind {
 				nodeList, err := serverClusterContext.KubernetesClient(t).CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
 				require.NoError(t, err)
 				// Get the address of the (only) node from the Kind cluster.
-				partitionSvcIP = nodeList.Items[0].Status.Addresses[0].Address
+				partitionSvcAddress = nodeList.Items[0].Status.Addresses[0].Address
+			} else {
+				// Get the IP of the partition service to configure the external server address in the values file for the clients cluster.
+				partitionServiceName := fmt.Sprintf("%s-consul-partition-service", releaseName)
+				logger.Logf(t, "retrieving partition service to determine external address for servers")
+				partitionsSvc, err := serverClusterContext.KubernetesClient(t).CoreV1().Services(serverClusterContext.KubectlOptions(t).Namespace).Get(ctx, partitionServiceName, metav1.GetOptions{})
+				require.NoError(t, err)
+
+				// On AWS, load balancers have a hostname for ingress, while on Azure and GCP
+				// load balancers have IPs.
+				if partitionsSvc.Status.LoadBalancer.Ingress[0].Hostname != "" {
+					partitionSvcAddress = partitionsSvc.Status.LoadBalancer.Ingress[0].Hostname
+				} else {
+					partitionSvcAddress = partitionsSvc.Status.LoadBalancer.Ingress[0].IP
+				}
 			}
 
 			var k8sAuthMethodHost string
-			// The Kubernetes AuthMethod IP for Kind is read from the endpoint for the Kubernetes service.
-			kubernetesEndpoint, err := clientClusterContext.KubernetesClient(t).CoreV1().Endpoints(defaultNamespace).Get(ctx, "kubernetes", metav1.GetOptions{})
-			require.NoError(t, err)
-			k8sAuthMethodHost = fmt.Sprintf("%s:%d", kubernetesEndpoint.Subsets[0].Addresses[0].IP, kubernetesEndpoint.Subsets[0].Ports[0].Port)
+			// When running on kind, the kube API address in kubeconfig will have a localhost address
+			// which will not work from inside the container. That's why we need to use the endpoints address instead
+			// which will point the node IP.
+			if cfg.UseKind {
+				// The Kubernetes AuthMethod host is read from the endpoints for the Kubernetes service.
+				kubernetesEndpoint, err := clientClusterContext.KubernetesClient(t).CoreV1().Endpoints(defaultNamespace).Get(ctx, "kubernetes", metav1.GetOptions{})
+				require.NoError(t, err)
+				k8sAuthMethodHost = fmt.Sprintf("%s:%d", kubernetesEndpoint.Subsets[0].Addresses[0].IP, kubernetesEndpoint.Subsets[0].Ports[0].Port)
+			} else {
+				k8sAuthMethodHost = helpers.KubernetesAPIServerHostFromOptions(t, clientClusterContext.KubectlOptions(t))
```

Comment:
yeah it looks like that worked for EKS, but on AKS I started seeing errors because the IP in the endpoints object was not in the certificate for the k8s api server. So I had to change it.